### PR TITLE
Add "API key" login/logout method

### PR DESCRIPTION
**Pull request by waxlamp**
This PR enables the use of "API keys" with multinetjs.

Note that this is largely a proof of concept. The names of the methods and the way in which I'm setting/unsetting the appropriate header in the axios object (and any other detail here) are up for discussion.

This feature is mainly to enable unit tests for this library. Note that the tests will work locally if you change the API key string that appears in the test, and will not yet work in CI at all. Once this approach is agreed upon and refined, and the Girder4 Multinet stack is deployed to testing, we can put in a CI secret the actual API key to use for testing.

@zachmullen, I'm tagging you here mainly so you can see how I'm approaching the auth problems I discussed with you yesterday. Please feel free to leave a review, or just general feedback--either or both would be appreciated.

Depends on #30.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -124,31 +130,76 @@ function fileToText(file: File): Promise<string> {
...
 }
 
-class MultinetAPI {
-  public client: Client;
+function extract<T>(promise: AxiosPromise<T>): Promise<T> {
+  return new Promise((resolve, reject) => {
+    promise.then((resp) => resolve(resp.data))
+      .catch((err) => reject(err));
+  });
+}
```
Comment (zachmullen) [inline src/index.ts:138]: Is the whole point of the above function just to extract the `.data` property from an object? Seems like that may not be necessary, and you could just destructure the result i.e. `const { data } = await ...`

Comment (waxlamp) [reply src/index.ts:138]: Yeah, I made the choice to do some extra work in crafting this library to save a small amount of cognitive load on application writers. Perhaps it's the wrong choice. Check out [this discussion](https://github.com/multinet-app/multinetjs/pull/30#discussion_r597148096) where Jeff brought up the same comment.

Comment (zachmullen) [reply src/index.ts:138]: I'd vote for removing this and just embracing our use of the axios API. Maybe a downstream will actually want to look at the HTTP status code of the response, or one of the headers. Dan Lamanna has convinced me of the virtue of avoiding "wrapping" a third party library with layers of abstraction that at best provide the same exact amount of power (but usually not as much) and also creating a lot of boilerplate code. To me, it's not a hard sell in terms of cognitive load: we just get to say that our API is essentially a superset of the axios API, which is already very popular, powerful, and well designed and documented.

Comment (waxlamp) [reply src/index.ts:138]: Now that you and Jeff have both brought up the exact same concern, I'm starting to think we should just simplify this. The change was actually instituted in that other PR, so I'd do the simplification there and rebase this PR appropriately.

@AlmightyYakob, any objection to taking Jeff and Zach's suggestion to heart?

**Comment by AlmightyYakob** [comment]
I believe in our API instance (default from Girder 4), endpoints end with a trailing slash, and accessing the non-trailing slash endpoint redirects to the former, so we'll want to make sure we add the trailing slashes into all of these endpoints.

**Comment by zachmullen** [comment]
The non-slash endpoints will redirect to the slash ones. FWIW on my projects I set `trailing_slash=False` because I hate them.

**Comment by JackWilb** [comment]
Closing this in favor of #77

**Pull request closed without merging**